src/controller/model/device.ts

```diff
@@ -705,16 +705,21 @@ class Device extends Entity {
 
         // Bind poll control
         try {
+            let retentionTimeout : number;
             for (const endpoint of this.endpoints.filter((e): boolean => e.supportsInputCluster('genPollCtrl'))) {
                 debug.log(`Interview - Poll control - binding '${this.ieeeAddr}' endpoint '${endpoint.ID}'`);
                 await endpoint.bind('genPollCtrl', coordinator.endpoints[0]);
-                const pollPeriod = await endpoint.read('genPollCtrl', ['checkinInterval']);
+                const pollPeriod = await endpoint.read('genPollCtrl', ['checkinInterval'], {sendWhen: 'immediate'});
+                retentionTimeout = pollPeriod.checkinInterval;
                 if (pollPeriod.checkinInterval <= 2400) {// 10 minutes
                     this.defaultSendRequestWhen = 'fastpoll';
                 } else {
                     this.defaultSendRequestWhen = 'active';
                 }
             }
+            this.endpoints.forEach((e) => {
```

Comment:
- Every device has only 1 checkinInterval, therefore there is no need to store a value per endpoint. I suggest to put this property on the device instead of endpoint. (same as `defaultSendRequestWhen`). You can provide a default value for this property `fromDatabaseEntry` (also similar to how `defaultSendRequestWhen` is hanlded). This reduces the need to check for existence of this property in the endpoint.
- I suggest to rename this to `pollCheckingInterval`. This value may be useful for other use cases and then `retentionTimeout` might be a confusing name.

Reply:
Thanks for the suggestions. I renamed and removed the variable as you suggested. I did not set the default value in `defaultSendRequestWhen`, however. I thought about this before and I decided not to do so, because that value would also be written back to the database later, where it becomes indistinguishable from actual device data. Since this is just a somewhat arbitrary value, I think this should not go to the database. What do you think?